In [112]:
#importing basics libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

In [113]:
df=pd.read_excel('data/flight_price (2).xlsx')
df.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

In [114]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10683 entries, 0 to 10682
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10683 non-null  object
 1   Date_of_Journey  10683 non-null  object
 2   Source           10683 non-null  object
 3   Destination      10683 non-null  object
 4   Route            10682 non-null  object
 5   Dep_Time         10683 non-null  object
 6   Arrival_Time     10683 non-null  object
 7   Duration         10683 non-null  object
 8   Total_Stops      10682 non-null  object
 9   Additional_Info  10683 non-null  object
 10  Price            10683 non-null  int64 
dtypes: int64(1), object(10)
memory usage: 918.2+ KB


In [115]:
df.nunique()

Airline              12
Date_of_Journey      44
Source                5
Destination           6
Route               128
Dep_Time            222
Arrival_Time       1343
Duration            368
Total_Stops           5
Additional_Info      10
Price              1870
dtype: int64

In [116]:
df.isna().sum()

Airline            0
Date_of_Journey    0
Source             0
Destination        0
Route              1
Dep_Time           0
Arrival_Time       0
Duration           0
Total_Stops        1
Additional_Info    0
Price              0
dtype: int64

In [117]:
df.isnull().sum()

Airline            0
Date_of_Journey    0
Source             0
Destination        0
Route              1
Dep_Time           0
Arrival_Time       0
Duration           0
Total_Stops        1
Additional_Info    0
Price              0
dtype: int64

### There is one null value in the Total_Stops column

In [118]:
df.shape

(10683, 11)

In [119]:
df.dropna(inplace=True)

In [120]:
df.shape

(10682, 11)

In [121]:
df.duplicated().sum()

220

### there are 220 duplated values in the data set

In [122]:
df.drop_duplicates(inplace=True)

In [123]:
df.shape

(10462, 11)

In [124]:
df.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

## Lets clean the columns one by one

In [125]:
df['Date']=df['Date_of_Journey'].str.split('/')

In [126]:
df.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  \
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897   
1    05:50         13:15   7h 25m     2 stops         No info   7662   
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882   
3    18:05         23:30   5h 25m      1 stop         No info   6218   
4    16:50         21:35   4h 45m      1 stop         No info  13302   

             Date  
0  [24, 03, 2019]  
1   [1, 05, 2019]  
2   [9, 06, 2019]  
3  [12, 05, 2019]  
4  [01, 03, 2019]

In [127]:
df['Date']=df['Date_of_Journey'].str.split('/').str[0]
df['Month']=df['Date_of_Journey'].str.split('/').str[1]
df['Year']=df['Date_of_Journey'].str.split('/').str[2]

In [128]:
df.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price Date  \
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897   24   
1    05:50         13:15   7h 25m     2 stops         No info   7662    1   
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882    9   
3    18:05         23:30   5h 25m      1 stop         No info   6218   12   
4    16:50         21:35   4h 45m      1 stop         No info  13302   01   

  Month  Year  
0    03  2019  
1    05  2019  
2    06  2019  
3    05  2019  
4    03  2019

In [129]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10462 entries, 0 to 10682
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10462 non-null  object
 1   Date_of_Journey  10462 non-null  object
 2   Source           10462 non-null  object
 3   Destination      10462 non-null  object
 4   Route            10462 non-null  object
 5   Dep_Time         10462 non-null  object
 6   Arrival_Time     10462 non-null  object
 7   Duration         10462 non-null  object
 8   Total_Stops      10462 non-null  object
 9   Additional_Info  10462 non-null  object
 10  Price            10462 non-null  int64 
 11  Date             10462 non-null  object
 12  Month            10462 non-null  object
 13  Year             10462 non-null  object
dtypes: int64(1), object(13)
memory usage: 1.2+ MB


In [130]:
df['Date']=df['Date'].astype(int)
df['Month']=df['Month'].astype(int)
df['Year']=df['Year'].astype(int)


In [131]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10462 entries, 0 to 10682
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10462 non-null  object
 1   Date_of_Journey  10462 non-null  object
 2   Source           10462 non-null  object
 3   Destination      10462 non-null  object
 4   Route            10462 non-null  object
 5   Dep_Time         10462 non-null  object
 6   Arrival_Time     10462 non-null  object
 7   Duration         10462 non-null  object
 8   Total_Stops      10462 non-null  object
 9   Additional_Info  10462 non-null  object
 10  Price            10462 non-null  int64 
 11  Date             10462 non-null  int32 
 12  Month            10462 non-null  int32 
 13  Year             10462 non-null  int32 
dtypes: int32(3), int64(1), object(10)
memory usage: 1.1+ MB


In [132]:
df.drop('Date_of_Journey',inplace=True,axis=1)

In [133]:
df.head()

Airline    Source Destination                  Route Dep_Time  \
0       IndiGo  Banglore   New Delhi              BLR → DEL    22:20   
1    Air India   Kolkata    Banglore  CCU → IXR → BBI → BLR    05:50   
2  Jet Airways     Delhi      Cochin  DEL → LKO → BOM → COK    09:25   
3       IndiGo   Kolkata    Banglore        CCU → NAG → BLR    18:05   
4       IndiGo  Banglore   New Delhi        BLR → NAG → DEL    16:50   

   Arrival_Time Duration Total_Stops Additional_Info  Price  Date  Month  Year  
0  01:10 22 Mar   2h 50m    non-stop         No info   3897    24      3  2019  
1         13:15   7h 25m     2 stops         No info   7662     1      5  2019  
2  04:25 10 Jun      19h     2 stops         No info  13882     9      6  2019  
3         23:30   5h 25m      1 stop         No info   6218    12      5  2019  
4         21:35   4h 45m      1 stop         No info  13302     1      3  2019

In [134]:
df['Airline'].unique()

array(['IndiGo', 'Air India', 'Jet Airways', 'SpiceJet',
       'Multiple carriers', 'GoAir', 'Vistara', 'Air Asia',
       'Vistara Premium economy', 'Jet Airways Business',
       'Multiple carriers Premium economy', 'Trujet'], dtype=object)

In [135]:
df['Destination'].unique()

array(['New Delhi', 'Banglore', 'Cochin', 'Kolkata', 'Delhi', 'Hyderabad'],
      dtype=object)

In [136]:
df['Source'].unique()

array(['Banglore', 'Kolkata', 'Delhi', 'Chennai', 'Mumbai'], dtype=object)

In [137]:
df['Total_Stops'].unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', '4 stops'],
      dtype=object)

In [138]:
df.drop('Route',axis=1,inplace=True)

In [139]:
df.head()

Airline    Source Destination Dep_Time  Arrival_Time Duration  \
0       IndiGo  Banglore   New Delhi    22:20  01:10 22 Mar   2h 50m   
1    Air India   Kolkata    Banglore    05:50         13:15   7h 25m   
2  Jet Airways     Delhi      Cochin    09:25  04:25 10 Jun      19h   
3       IndiGo   Kolkata    Banglore    18:05         23:30   5h 25m   
4       IndiGo  Banglore   New Delhi    16:50         21:35   4h 45m   

  Total_Stops Additional_Info  Price  Date  Month  Year  
0    non-stop         No info   3897    24      3  2019  
1     2 stops         No info   7662     1      5  2019  
2     2 stops         No info  13882     9      6  2019  
3      1 stop         No info   6218    12      5  2019  
4      1 stop         No info  13302     1      3  2019

## cleaning the Dep_Time column

In [140]:
df['dep_hour']=df['Dep_Time'].str.split(':').str[0]
df['dep_min']=df['Dep_Time'].str.split(':').str[1]

In [141]:
df.head()

Airline    Source Destination Dep_Time  Arrival_Time Duration  \
0       IndiGo  Banglore   New Delhi    22:20  01:10 22 Mar   2h 50m   
1    Air India   Kolkata    Banglore    05:50         13:15   7h 25m   
2  Jet Airways     Delhi      Cochin    09:25  04:25 10 Jun      19h   
3       IndiGo   Kolkata    Banglore    18:05         23:30   5h 25m   
4       IndiGo  Banglore   New Delhi    16:50         21:35   4h 45m   

  Total_Stops Additional_Info  Price  Date  Month  Year dep_hour dep_min  
0    non-stop         No info   3897    24      3  2019       22      20  
1     2 stops         No info   7662     1      5  2019       05      50  
2     2 stops         No info  13882     9      6  2019       09      25  
3      1 stop         No info   6218    12      5  2019       18      05  
4      1 stop         No info  13302     1      3  2019       16      50

In [142]:
df.drop('Dep_Time',inplace=True,axis=1)

In [143]:
df.head()

Airline    Source Destination  Arrival_Time Duration Total_Stops  \
0       IndiGo  Banglore   New Delhi  01:10 22 Mar   2h 50m    non-stop   
1    Air India   Kolkata    Banglore         13:15   7h 25m     2 stops   
2  Jet Airways     Delhi      Cochin  04:25 10 Jun      19h     2 stops   
3       IndiGo   Kolkata    Banglore         23:30   5h 25m      1 stop   
4       IndiGo  Banglore   New Delhi         21:35   4h 45m      1 stop   

  Additional_Info  Price  Date  Month  Year dep_hour dep_min  
0         No info   3897    24      3  2019       22      20  
1         No info   7662     1      5  2019       05      50  
2         No info  13882     9      6  2019       09      25  
3         No info   6218    12      5  2019       18      05  
4         No info  13302     1      3  2019       16      50

In [144]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10462 entries, 0 to 10682
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10462 non-null  object
 1   Source           10462 non-null  object
 2   Destination      10462 non-null  object
 3   Arrival_Time     10462 non-null  object
 4   Duration         10462 non-null  object
 5   Total_Stops      10462 non-null  object
 6   Additional_Info  10462 non-null  object
 7   Price            10462 non-null  int64 
 8   Date             10462 non-null  int32 
 9   Month            10462 non-null  int32 
 10  Year             10462 non-null  int32 
 11  dep_hour         10462 non-null  object
 12  dep_min          10462 non-null  object
dtypes: int32(3), int64(1), object(9)
memory usage: 1021.7+ KB


In [145]:
df['dep_hour']=df['dep_hour'].astype(int)
df['dep_min']=df['dep_min'].astype(int)

In [146]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10462 entries, 0 to 10682
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10462 non-null  object
 1   Source           10462 non-null  object
 2   Destination      10462 non-null  object
 3   Arrival_Time     10462 non-null  object
 4   Duration         10462 non-null  object
 5   Total_Stops      10462 non-null  object
 6   Additional_Info  10462 non-null  object
 7   Price            10462 non-null  int64 
 8   Date             10462 non-null  int32 
 9   Month            10462 non-null  int32 
 10  Year             10462 non-null  int32 
 11  dep_hour         10462 non-null  int32 
 12  dep_min          10462 non-null  int32 
dtypes: int32(5), int64(1), object(7)
memory usage: 939.9+ KB


## cleaning the Arrival Time column

In [147]:
df.head()

Airline    Source Destination  Arrival_Time Duration Total_Stops  \
0       IndiGo  Banglore   New Delhi  01:10 22 Mar   2h 50m    non-stop   
1    Air India   Kolkata    Banglore         13:15   7h 25m     2 stops   
2  Jet Airways     Delhi      Cochin  04:25 10 Jun      19h     2 stops   
3       IndiGo   Kolkata    Banglore         23:30   5h 25m      1 stop   
4       IndiGo  Banglore   New Delhi         21:35   4h 45m      1 stop   

  Additional_Info  Price  Date  Month  Year  dep_hour  dep_min  
0         No info   3897    24      3  2019        22       20  
1         No info   7662     1      5  2019         5       50  
2         No info  13882     9      6  2019         9       25  
3         No info   6218    12      5  2019        18        5  
4         No info  13302     1      3  2019        16       50

In [148]:
df['Arrival_hour']=df['Arrival_Time'].str.split(':').str[0]
df['arrival_min']=df['Arrival_Time'].str.split(':').str[1]
df['Arrival_min']=df['arrival_min'].str.split(' ').str[0]
df.drop('Arrival_Time',inplace=True,axis=1)



In [149]:
df.head()

Airline    Source Destination Duration Total_Stops Additional_Info  \
0       IndiGo  Banglore   New Delhi   2h 50m    non-stop         No info   
1    Air India   Kolkata    Banglore   7h 25m     2 stops         No info   
2  Jet Airways     Delhi      Cochin      19h     2 stops         No info   
3       IndiGo   Kolkata    Banglore   5h 25m      1 stop         No info   
4       IndiGo  Banglore   New Delhi   4h 45m      1 stop         No info   

   Price  Date  Month  Year  dep_hour  dep_min Arrival_hour arrival_min  \
0   3897    24      3  2019        22       20           01   10 22 Mar   
1   7662     1      5  2019         5       50           13          15   
2  13882     9      6  2019         9       25           04   25 10 Jun   
3   6218    12      5  2019        18        5           23          30   
4  13302     1      3  2019        16       50           21          35   

  Arrival_min  
0          10  
1          15  
2          25  
3          30  
4          35

In [150]:
df.drop('arrival_min',inplace=True,axis=1)

In [151]:
df.head()

Airline    Source Destination Duration Total_Stops Additional_Info  \
0       IndiGo  Banglore   New Delhi   2h 50m    non-stop         No info   
1    Air India   Kolkata    Banglore   7h 25m     2 stops         No info   
2  Jet Airways     Delhi      Cochin      19h     2 stops         No info   
3       IndiGo   Kolkata    Banglore   5h 25m      1 stop         No info   
4       IndiGo  Banglore   New Delhi   4h 45m      1 stop         No info   

   Price  Date  Month  Year  dep_hour  dep_min Arrival_hour Arrival_min  
0   3897    24      3  2019        22       20           01          10  
1   7662     1      5  2019         5       50           13          15  
2  13882     9      6  2019         9       25           04          25  
3   6218    12      5  2019        18        5           23          30  
4  13302     1      3  2019        16       50           21          35

In [152]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10462 entries, 0 to 10682
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10462 non-null  object
 1   Source           10462 non-null  object
 2   Destination      10462 non-null  object
 3   Duration         10462 non-null  object
 4   Total_Stops      10462 non-null  object
 5   Additional_Info  10462 non-null  object
 6   Price            10462 non-null  int64 
 7   Date             10462 non-null  int32 
 8   Month            10462 non-null  int32 
 9   Year             10462 non-null  int32 
 10  dep_hour         10462 non-null  int32 
 11  dep_min          10462 non-null  int32 
 12  Arrival_hour     10462 non-null  object
 13  Arrival_min      10462 non-null  object
dtypes: int32(5), int64(1), object(8)
memory usage: 1021.7+ KB


In [153]:
df['Arrival_hour']=df['Arrival_hour'].astype(int)
df['Arrival_min']=df['Arrival_min'].astype(int)

In [154]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10462 entries, 0 to 10682
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10462 non-null  object
 1   Source           10462 non-null  object
 2   Destination      10462 non-null  object
 3   Duration         10462 non-null  object
 4   Total_Stops      10462 non-null  object
 5   Additional_Info  10462 non-null  object
 6   Price            10462 non-null  int64 
 7   Date             10462 non-null  int32 
 8   Month            10462 non-null  int32 
 9   Year             10462 non-null  int32 
 10  dep_hour         10462 non-null  int32 
 11  dep_min          10462 non-null  int32 
 12  Arrival_hour     10462 non-null  int32 
 13  Arrival_min      10462 non-null  int32 
dtypes: int32(7), int64(1), object(6)
memory usage: 939.9+ KB


In [155]:
df.head()

Airline    Source Destination Duration Total_Stops Additional_Info  \
0       IndiGo  Banglore   New Delhi   2h 50m    non-stop         No info   
1    Air India   Kolkata    Banglore   7h 25m     2 stops         No info   
2  Jet Airways     Delhi      Cochin      19h     2 stops         No info   
3       IndiGo   Kolkata    Banglore   5h 25m      1 stop         No info   
4       IndiGo  Banglore   New Delhi   4h 45m      1 stop         No info   

   Price  Date  Month  Year  dep_hour  dep_min  Arrival_hour  Arrival_min  
0   3897    24      3  2019        22       20             1           10  
1   7662     1      5  2019         5       50            13           15  
2  13882     9      6  2019         9       25             4           25  
3   6218    12      5  2019        18        5            23           30  
4  13302     1      3  2019        16       50            21           35

## Cleaning the Duration column

In [156]:
df['duration_hour']=df['Duration'].str.split(' ').str[0]
df['duration_min']=df['Duration'].str.split(' ').str[1]

In [157]:
df.head()

Airline    Source Destination Duration Total_Stops Additional_Info  \
0       IndiGo  Banglore   New Delhi   2h 50m    non-stop         No info   
1    Air India   Kolkata    Banglore   7h 25m     2 stops         No info   
2  Jet Airways     Delhi      Cochin      19h     2 stops         No info   
3       IndiGo   Kolkata    Banglore   5h 25m      1 stop         No info   
4       IndiGo  Banglore   New Delhi   4h 45m      1 stop         No info   

   Price  Date  Month  Year  dep_hour  dep_min  Arrival_hour  Arrival_min  \
0   3897    24      3  2019        22       20             1           10   
1   7662     1      5  2019         5       50            13           15   
2  13882     9      6  2019         9       25             4           25   
3   6218    12      5  2019        18        5            23           30   
4  13302     1      3  2019        16       50            21           35   

  duration_hour duration_min  
0            2h          50m  
1            7h          25m  
2           19h          NaN  
3            5h          25m  
4            4h          45m

In [158]:
df['Duration_min']=df['duration_min'].fillna('0m')

In [159]:
df.head()

Airline    Source Destination Duration Total_Stops Additional_Info  \
0       IndiGo  Banglore   New Delhi   2h 50m    non-stop         No info   
1    Air India   Kolkata    Banglore   7h 25m     2 stops         No info   
2  Jet Airways     Delhi      Cochin      19h     2 stops         No info   
3       IndiGo   Kolkata    Banglore   5h 25m      1 stop         No info   
4       IndiGo  Banglore   New Delhi   4h 45m      1 stop         No info   

   Price  Date  Month  Year  dep_hour  dep_min  Arrival_hour  Arrival_min  \
0   3897    24      3  2019        22       20             1           10   
1   7662     1      5  2019         5       50            13           15   
2  13882     9      6  2019         9       25             4           25   
3   6218    12      5  2019        18        5            23           30   
4  13302     1      3  2019        16       50            21           35   

  duration_hour duration_min Duration_min  
0            2h          50m          50m  
1            7h          25m          25m  
2           19h          NaN           0m  
3            5h          25m          25m  
4            4h          45m          45m

In [160]:
df.drop('duration_min',inplace=True,axis=1)

In [161]:
df.head()

Airline    Source Destination Duration Total_Stops Additional_Info  \
0       IndiGo  Banglore   New Delhi   2h 50m    non-stop         No info   
1    Air India   Kolkata    Banglore   7h 25m     2 stops         No info   
2  Jet Airways     Delhi      Cochin      19h     2 stops         No info   
3       IndiGo   Kolkata    Banglore   5h 25m      1 stop         No info   
4       IndiGo  Banglore   New Delhi   4h 45m      1 stop         No info   

   Price  Date  Month  Year  dep_hour  dep_min  Arrival_hour  Arrival_min  \
0   3897    24      3  2019        22       20             1           10   
1   7662     1      5  2019         5       50            13           15   
2  13882     9      6  2019         9       25             4           25   
3   6218    12      5  2019        18        5            23           30   
4  13302     1      3  2019        16       50            21           35   

  duration_hour Duration_min  
0            2h          50m  
1            7h          25m  
2           19h           0m  
3            5h          25m  
4            4h          45m

In [162]:
df['Duration_min'].unique()

array(['50m', '25m', '0m', '45m', '30m', '5m', '15m', '35m', '10m', '20m',
       '55m', '40m'], dtype=object)

In [163]:
df['duration_hr']=df['duration_hour'].str.split('h').str[0]
df['duration_min']=df['Duration_min'].str.split('m').str[0]

In [164]:
df.head()

Airline    Source Destination Duration Total_Stops Additional_Info  \
0       IndiGo  Banglore   New Delhi   2h 50m    non-stop         No info   
1    Air India   Kolkata    Banglore   7h 25m     2 stops         No info   
2  Jet Airways     Delhi      Cochin      19h     2 stops         No info   
3       IndiGo   Kolkata    Banglore   5h 25m      1 stop         No info   
4       IndiGo  Banglore   New Delhi   4h 45m      1 stop         No info   

   Price  Date  Month  Year  dep_hour  dep_min  Arrival_hour  Arrival_min  \
0   3897    24      3  2019        22       20             1           10   
1   7662     1      5  2019         5       50            13           15   
2  13882     9      6  2019         9       25             4           25   
3   6218    12      5  2019        18        5            23           30   
4  13302     1      3  2019        16       50            21           35   

  duration_hour Duration_min duration_hr duration_min  
0            2h          50m           2           50  
1            7h          25m           7           25  
2           19h           0m          19            0  
3            5h          25m           5           25  
4            4h          45m           4           45

In [165]:
df['duration_hr'].unique()


array(['2', '7', '19', '5', '4', '15', '21', '25', '13', '12', '26', '22',
       '23', '20', '10', '6', '11', '8', '16', '3', '27', '1', '14', '9',
       '18', '17', '24', '30', '28', '29', '37', '34', '38', '35', '36',
       '47', '33', '32', '31', '42', '39', '5m', '41', '40'], dtype=object)

In [166]:
df['duration_min'].unique()

array(['50', '25', '0', '45', '30', '5', '15', '35', '10', '20', '55',
       '40'], dtype=object)

In [167]:
df.drop('duration_hour',inplace=True,axis=1)
df.drop('Duration_min',inplace=True,axis=1)

In [168]:
df.head()

Airline    Source Destination Duration Total_Stops Additional_Info  \
0       IndiGo  Banglore   New Delhi   2h 50m    non-stop         No info   
1    Air India   Kolkata    Banglore   7h 25m     2 stops         No info   
2  Jet Airways     Delhi      Cochin      19h     2 stops         No info   
3       IndiGo   Kolkata    Banglore   5h 25m      1 stop         No info   
4       IndiGo  Banglore   New Delhi   4h 45m      1 stop         No info   

   Price  Date  Month  Year  dep_hour  dep_min  Arrival_hour  Arrival_min  \
0   3897    24      3  2019        22       20             1           10   
1   7662     1      5  2019         5       50            13           15   
2  13882     9      6  2019         9       25             4           25   
3   6218    12      5  2019        18        5            23           30   
4  13302     1      3  2019        16       50            21           35   

  duration_hr duration_min  
0           2           50  
1           7           25  
2          19            0  
3           5           25  
4           4           45

In [169]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10462 entries, 0 to 10682
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10462 non-null  object
 1   Source           10462 non-null  object
 2   Destination      10462 non-null  object
 3   Duration         10462 non-null  object
 4   Total_Stops      10462 non-null  object
 5   Additional_Info  10462 non-null  object
 6   Price            10462 non-null  int64 
 7   Date             10462 non-null  int32 
 8   Month            10462 non-null  int32 
 9   Year             10462 non-null  int32 
 10  dep_hour         10462 non-null  int32 
 11  dep_min          10462 non-null  int32 
 12  Arrival_hour     10462 non-null  int32 
 13  Arrival_min      10462 non-null  int32 
 14  duration_hr      10462 non-null  object
 15  duration_min     10462 non-null  object
dtypes: int32(7), int64(1), object(8)
memory usage: 1.1+ MB


In [170]:
df['duration_min']=df['duration_min'].astype(int)
# df['duration_hr']=df['duration_hr'].astype(int)

In [171]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10462 entries, 0 to 10682
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10462 non-null  object
 1   Source           10462 non-null  object
 2   Destination      10462 non-null  object
 3   Duration         10462 non-null  object
 4   Total_Stops      10462 non-null  object
 5   Additional_Info  10462 non-null  object
 6   Price            10462 non-null  int64 
 7   Date             10462 non-null  int32 
 8   Month            10462 non-null  int32 
 9   Year             10462 non-null  int32 
 10  dep_hour         10462 non-null  int32 
 11  dep_min          10462 non-null  int32 
 12  Arrival_hour     10462 non-null  int32 
 13  Arrival_min      10462 non-null  int32 
 14  duration_hr      10462 non-null  object
 15  duration_min     10462 non-null  int32 
dtypes: int32(8), int64(1), object(7)
memory usage: 1.0+ MB


In [172]:
df.head()

Airline    Source Destination Duration Total_Stops Additional_Info  \
0       IndiGo  Banglore   New Delhi   2h 50m    non-stop         No info   
1    Air India   Kolkata    Banglore   7h 25m     2 stops         No info   
2  Jet Airways     Delhi      Cochin      19h     2 stops         No info   
3       IndiGo   Kolkata    Banglore   5h 25m      1 stop         No info   
4       IndiGo  Banglore   New Delhi   4h 45m      1 stop         No info   

   Price  Date  Month  Year  dep_hour  dep_min  Arrival_hour  Arrival_min  \
0   3897    24      3  2019        22       20             1           10   
1   7662     1      5  2019         5       50            13           15   
2  13882     9      6  2019         9       25             4           25   
3   6218    12      5  2019        18        5            23           30   
4  13302     1      3  2019        16       50            21           35   

  duration_hr  duration_min  
0           2            50  
1           7            25  
2          19             0  
3           5            25  
4           4            45

In [173]:
df.drop('Duration',axis=1,inplace=True)

In [174]:
df.head()

Airline    Source Destination Total_Stops Additional_Info  Price  Date  \
0       IndiGo  Banglore   New Delhi    non-stop         No info   3897    24   
1    Air India   Kolkata    Banglore     2 stops         No info   7662     1   
2  Jet Airways     Delhi      Cochin     2 stops         No info  13882     9   
3       IndiGo   Kolkata    Banglore      1 stop         No info   6218    12   
4       IndiGo  Banglore   New Delhi      1 stop         No info  13302     1   

   Month  Year  dep_hour  dep_min  Arrival_hour  Arrival_min duration_hr  \
0      3  2019        22       20             1           10           2   
1      5  2019         5       50            13           15           7   
2      6  2019         9       25             4           25          19   
3      5  2019        18        5            23           30           5   
4      3  2019        16       50            21           35           4   

   duration_min  
0            50  
1            25  
2             0  
3            25  
4            45

## cleaning the Total_Stops column

In [175]:
df['Total_Stops'].unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', '4 stops'],
      dtype=object)

### map function is used for data encoding

In [176]:
# df['Total_Stops']=df['Total_Stops'].map({'non-stop':0, '2 stops':2, '1 stop':1, '3 stops':3, '4 stops':4})

In [177]:
df['Total_Stops'].unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', '4 stops'],
      dtype=object)

In [178]:
df.head()

Airline    Source Destination Total_Stops Additional_Info  Price  Date  \
0       IndiGo  Banglore   New Delhi    non-stop         No info   3897    24   
1    Air India   Kolkata    Banglore     2 stops         No info   7662     1   
2  Jet Airways     Delhi      Cochin     2 stops         No info  13882     9   
3       IndiGo   Kolkata    Banglore      1 stop         No info   6218    12   
4       IndiGo  Banglore   New Delhi      1 stop         No info  13302     1   

   Month  Year  dep_hour  dep_min  Arrival_hour  Arrival_min duration_hr  \
0      3  2019        22       20             1           10           2   
1      5  2019         5       50            13           15           7   
2      6  2019         9       25             4           25          19   
3      5  2019        18        5            23           30           5   
4      3  2019        16       50            21           35           4   

   duration_min  
0            50  
1            25  
2             0  
3            25  
4            45

In [179]:
df.drop('Additional_Info',inplace=True,axis=1)

In [180]:
df.head()

Airline    Source Destination Total_Stops  Price  Date  Month  Year  \
0       IndiGo  Banglore   New Delhi    non-stop   3897    24      3  2019   
1    Air India   Kolkata    Banglore     2 stops   7662     1      5  2019   
2  Jet Airways     Delhi      Cochin     2 stops  13882     9      6  2019   
3       IndiGo   Kolkata    Banglore      1 stop   6218    12      5  2019   
4       IndiGo  Banglore   New Delhi      1 stop  13302     1      3  2019   

   dep_hour  dep_min  Arrival_hour  Arrival_min duration_hr  duration_min  
0        22       20             1           10           2            50  
1         5       50            13           15           7            25  
2         9       25             4           25          19             0  
3        18        5            23           30           5            25  
4        16       50            21           35           4            45

In [181]:
df['duration_hr'].unique()

array(['2', '7', '19', '5', '4', '15', '21', '25', '13', '12', '26', '22',
       '23', '20', '10', '6', '11', '8', '16', '3', '27', '1', '14', '9',
       '18', '17', '24', '30', '28', '29', '37', '34', '38', '35', '36',
       '47', '33', '32', '31', '42', '39', '5m', '41', '40'], dtype=object)

In [182]:
# Drop rows where 'values' is '5m'
df = df[df['duration_hr'] != '5m']


In [183]:
df.reset_index(drop=True, inplace=True)


In [184]:
df['duration_hr'].unique()

array(['2', '7', '19', '5', '4', '15', '21', '25', '13', '12', '26', '22',
       '23', '20', '10', '6', '11', '8', '16', '3', '27', '1', '14', '9',
       '18', '17', '24', '30', '28', '29', '37', '34', '38', '35', '36',
       '47', '33', '32', '31', '42', '39', '41', '40'], dtype=object)

In [185]:
df['duration_hr']=df['duration_hr'].astype(int)


In [186]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10461 entries, 0 to 10460
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Airline       10461 non-null  object
 1   Source        10461 non-null  object
 2   Destination   10461 non-null  object
 3   Total_Stops   10461 non-null  object
 4   Price         10461 non-null  int64 
 5   Date          10461 non-null  int32 
 6   Month         10461 non-null  int32 
 7   Year          10461 non-null  int32 
 8   dep_hour      10461 non-null  int32 
 9   dep_min       10461 non-null  int32 
 10  Arrival_hour  10461 non-null  int32 
 11  Arrival_min   10461 non-null  int32 
 12  duration_hr   10461 non-null  int32 
 13  duration_min  10461 non-null  int32 
dtypes: int32(9), int64(1), object(4)
memory usage: 776.5+ KB


In [187]:
df['Airline'].unique()

array(['IndiGo', 'Air India', 'Jet Airways', 'SpiceJet',
       'Multiple carriers', 'GoAir', 'Vistara', 'Air Asia',
       'Vistara Premium economy', 'Jet Airways Business',
       'Multiple carriers Premium economy', 'Trujet'], dtype=object)

In [188]:
df.head(1)

Airline    Source Destination Total_Stops  Price  Date  Month  Year  \
0  IndiGo  Banglore   New Delhi    non-stop   3897    24      3  2019   

   dep_hour  dep_min  Arrival_hour  Arrival_min  duration_hr  duration_min  
0        22       20             1           10            2            50

In [189]:
df.drop('Year',inplace=True,axis=1)
df.drop('dep_hour',inplace=True,axis=1)
df.drop('dep_min',inplace=True,axis=1)
df.drop('Arrival_hour',inplace=True,axis=1)
df.drop('Arrival_min',inplace=True,axis=1)
df.drop('duration_hr',inplace=True,axis=1)
df.drop('duration_min',inplace=True,axis=1)


In [190]:
df.head(1)

Airline    Source Destination Total_Stops  Price  Date  Month
0  IndiGo  Banglore   New Delhi    non-stop   3897    24      3

In [192]:
df.columns

Index(['Airline', 'Source', 'Destination', 'Total_Stops', 'Price', 'Date',
       'Month'],
      dtype='object')

In [85]:
df.to_csv('cleaned_data1.csv',index=True)

## Second Cleaning Part

In [209]:
import pandas as pd

df1=pd.read_excel('data/flight_price (2).xlsx')
df1.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

In [211]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10683 entries, 0 to 10682
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10683 non-null  object
 1   Date_of_Journey  10683 non-null  object
 2   Source           10683 non-null  object
 3   Destination      10683 non-null  object
 4   Route            10682 non-null  object
 5   Dep_Time         10683 non-null  object
 6   Arrival_Time     10683 non-null  object
 7   Duration         10683 non-null  object
 8   Total_Stops      10682 non-null  object
 9   Additional_Info  10683 non-null  object
 10  Price            10683 non-null  int64 
dtypes: int64(1), object(10)
memory usage: 918.2+ KB


In [212]:
df1.nunique()

Airline              12
Date_of_Journey      44
Source                5
Destination           6
Route               128
Dep_Time            222
Arrival_Time       1343
Duration            368
Total_Stops           5
Additional_Info      10
Price              1870
dtype: int64

In [213]:
df1.isna().sum()

Airline            0
Date_of_Journey    0
Source             0
Destination        0
Route              1
Dep_Time           0
Arrival_Time       0
Duration           0
Total_Stops        1
Additional_Info    0
Price              0
dtype: int64

In [214]:
df1.isnull().sum()

Airline            0
Date_of_Journey    0
Source             0
Destination        0
Route              1
Dep_Time           0
Arrival_Time       0
Duration           0
Total_Stops        1
Additional_Info    0
Price              0
dtype: int64

In [215]:
df1.shape

(10683, 11)

In [216]:
df1.dropna(inplace=True)

In [220]:
df1.duplicated().sum()

220

In [221]:
df1.drop_duplicates(inplace=True)

In [222]:
df1.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

In [223]:
df1['Date']=df1['Date_of_Journey'].str.split('/')

In [224]:
df1.head(1)

Airline Date_of_Journey    Source Destination      Route Dep_Time  \
0  IndiGo      24/03/2019  Banglore   New Delhi  BLR → DEL    22:20   

   Arrival_Time Duration Total_Stops Additional_Info  Price            Date  
0  01:10 22 Mar   2h 50m    non-stop         No info   3897  [24, 03, 2019]

In [225]:
df1['Date']=df1['Date_of_Journey'].str.split('/').str[0]
df1['Month']=df1['Date_of_Journey'].str.split('/').str[1]
df1['Year']=df1['Date_of_Journey'].str.split('/').str[2]

In [226]:
df1.head(2)

Airline Date_of_Journey    Source Destination                  Route  \
0     IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1  Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price Date  \
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897   24   
1    05:50         13:15   7h 25m     2 stops         No info   7662    1   

  Month  Year  
0    03  2019  
1    05  2019

In [228]:
df1['Date']=df1['Date'].astype(int)
df1['Month']=df1['Month'].astype(int)
df1['Year']=df1['Year'].astype(int)

In [229]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10462 entries, 0 to 10682
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10462 non-null  object
 1   Date_of_Journey  10462 non-null  object
 2   Source           10462 non-null  object
 3   Destination      10462 non-null  object
 4   Route            10462 non-null  object
 5   Dep_Time         10462 non-null  object
 6   Arrival_Time     10462 non-null  object
 7   Duration         10462 non-null  object
 8   Total_Stops      10462 non-null  object
 9   Additional_Info  10462 non-null  object
 10  Price            10462 non-null  int64 
 11  Date             10462 non-null  int32 
 12  Month            10462 non-null  int32 
 13  Year             10462 non-null  int32 
dtypes: int32(3), int64(1), object(10)
memory usage: 1.1+ MB


In [230]:
df1.head(2)

Airline Date_of_Journey    Source Destination                  Route  \
0     IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1  Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  Date  \
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897    24   
1    05:50         13:15   7h 25m     2 stops         No info   7662     1   

   Month  Year  
0      3  2019  
1      5  2019

In [231]:
df1.drop('Date_of_Journey',inplace=True,axis=1)
df1.drop('Route',inplace=True,axis=1)
df1.drop('Dep_Time',inplace=True,axis=1)
df1.drop('Arrival_Time',inplace=True,axis=1)
df1.drop('Duration',inplace=True,axis=1)
df1.drop('Additional_Info',inplace=True,axis=1)
df1.drop('Year',inplace=True,axis=1)


In [232]:
df1.head()

Airline    Source Destination Total_Stops  Price  Date  Month
0       IndiGo  Banglore   New Delhi    non-stop   3897    24      3
1    Air India   Kolkata    Banglore     2 stops   7662     1      5
2  Jet Airways     Delhi      Cochin     2 stops  13882     9      6
3       IndiGo   Kolkata    Banglore      1 stop   6218    12      5
4       IndiGo  Banglore   New Delhi      1 stop  13302     1      3

In [237]:
df1['Total_Stops'].unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', '4 stops'],
      dtype=object)

In [235]:
df1['Destination'].unique()

array(['New Delhi', 'Banglore', 'Cochin', 'Kolkata', 'Delhi', 'Hyderabad'],
      dtype=object)

In [234]:
df1['Source'].unique()

array(['Banglore', 'Kolkata', 'Delhi', 'Chennai', 'Mumbai'], dtype=object)

In [233]:
df1['Airline'].unique()

array(['IndiGo', 'Air India', 'Jet Airways', 'SpiceJet',
       'Multiple carriers', 'GoAir', 'Vistara', 'Air Asia',
       'Vistara Premium economy', 'Jet Airways Business',
       'Multiple carriers Premium economy', 'Trujet'], dtype=object)

In [238]:
df1['Date'].unique()

array([24,  1,  9, 12, 27, 18,  3, 15,  6, 21])

In [239]:
df1['Month'].unique()

array([3, 5, 6, 4])

In [242]:
df1.isnull().sum()

Airline        0
Source         0
Destination    0
Total_Stops    0
Price          0
Date           0
Month          0
dtype: int64

In [243]:
df1.to_csv('cleaned_data2.csv',index=True)

In [208]:
df['Airline'].unique()

array([ 4,  6, 10,  3,  8,  2,  7,  1,  9, 12, 11,  5], dtype=int64)

In [109]:
Airline_map={"Air Asia":1,"GoAir":2,"SpiceJet":3,"IndiGo":4,"Trujet":5,"Air India":6,"Vistara":7,"Multiple carriers":8,"Vistara Premium economy":9,"Jet Airways":10,"Multiple carriers Premium economy":11,"Jet Airways Business":12}

In [110]:
Airline_map

{'Air Asia': 1,
 'GoAir': 2,
 'SpiceJet': 3,
 'IndiGo': 4,
 'Trujet': 5,
 'Air India': 6,
 'Vistara': 7,
 'Multiple carriers': 8,
 'Vistara Premium economy': 9,
 'Jet Airways': 10,
 'Multiple carriers Premium economy': 11,
 'Jet Airways Business': 12}

In [193]:
df['Source'].unique()

array(['Banglore', 'Kolkata', 'Delhi', 'Chennai', 'Mumbai'], dtype=object)

In [194]:
Source_map={"Delhi":1,"Mumbai":2,"Banglore":3,"Kolkata":4,"Chennai":5}

In [195]:
Source_map

{'Delhi': 1, 'Mumbai': 2, 'Banglore': 3, 'Kolkata': 4, 'Chennai': 5}

In [196]:
df['Destination'].unique()

array(['New Delhi', 'Banglore', 'Cochin', 'Kolkata', 'Delhi', 'Hyderabad'],
      dtype=object)

In [197]:
Destination_map={"Delhi":1,"New Delhi":2,"Banglore":3,"Kolkata":4,"Hyderabad":5,"Cochin":6}

In [198]:
Destination_map

{'Delhi': 1,
 'New Delhi': 2,
 'Banglore': 3,
 'Kolkata': 4,
 'Hyderabad': 5,
 'Cochin': 6}

In [199]:
df['Total_Stops'].unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', '4 stops'],
      dtype=object)

In [200]:
Stops_map={"non-stop":1,"1 stop":2,"2 stops":3,"3 stops":4,"4 stops":5}

In [201]:
Stops_map

{'non-stop': 1, '1 stop': 2, '2 stops': 3, '3 stops': 4, '4 stops': 5}

In [202]:
df.head(1)

Airline    Source Destination Total_Stops  Price  Date  Month
0  IndiGo  Banglore   New Delhi    non-stop   3897    24      3

In [203]:
df['Airline']=df['Airline'].map(Airline_map)
df['Source']=df['Source'].map(Source_map)
df['Destination']=df['Destination'].map(Destination_map)
df['Total_Stops']=df['Total_Stops'].map(Stops_map)


In [205]:
df.head()

Airline  Source  Destination  Total_Stops  Price  Date  Month
0        4       3            2            1   3897    24      3
1        6       4            3            3   7662     1      5
2       10       1            6            3  13882     9      6
3        4       4            3            2   6218    12      5
4        4       3            2            2  13302     1      3

In [206]:
df.to_csv('cleaned_data2.csv',index=True)

In [207]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10461 entries, 0 to 10460
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   Airline      10461 non-null  int64
 1   Source       10461 non-null  int64
 2   Destination  10461 non-null  int64
 3   Total_Stops  10461 non-null  int64
 4   Price        10461 non-null  int64
 5   Date         10461 non-null  int32
 6   Month        10461 non-null  int32
dtypes: int32(2), int64(5)
memory usage: 490.5 KB
